contracts/basset_vault/src/state.rs

```diff
@@ -53,6 +51,25 @@ pub struct GovernanceUpdateState {
     pub wait_approve_until: u64,
 }
 
+/// Data used only to init psi distrubutor
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct PsiDistributorInitInfo {
+    /// Terraswap factory contract to create nAsset <-> Psi pair
+    pub terraswap_factory_contract_addr: String,
+    /// Address of nAsset <-> Psi pair.
+    /// It is optional because it is set later,
+    /// after pair was created using `terraswap_factory_contract_addr`
+    pub nasset_psi_swap_contract_addr: Option<String>,
+}
+
+impl PsiDistributorInitInfo {
+    /// Convenient way to get `nasset_psi_swap_contract_addr` wrapped in `StdResult`
+    pub fn into_nasset_psi_swap_contract_addr(self) -> StdResult<String> {
```

Comment:
How about rename it to `get_nasset_psi_swap_contract_addr`?

Reply:
Or you did it cause of `self`?

Reply:
Yes, it consumes `self` to return internal `String`, so according to rust naming conventions it should be `into`

Reply:
ok